textile/chat-features.textile

```diff
@@ -9,5 +9,495 @@ jump_to:
+*** @(CHA-RL2h2)@ @[Testable]@ If the room is already in a @FAILED@ status during the detach operation and another channel fails, the @FAILED@ status will not be emitted twice.
+*** @(CHA-RL2h3)@ @[Testable]@ If a channel enters another state during detachment (namely @ATTACHED@, which happens when detach times out). Then the detachment cycle will be retried after a short pause. The rooms status will not change during this process.
+* @(CHA-RL3)@ A room must be explicitly released via the @RELEASE@ operation.. This operation is not performed directly on a Room object by the client, but is described here.
+** @(CHA-RL3a)@ @[Testable]@ If the room is already in the @RELEASED@ status, this operation is no-op.
+** @(CHA-RL3b)@ @[Testable]@ If the room is in the @DETACHED@ status, then the room is immediately transitioned to @RELEASED@ and the operation succeeds.
+** @(CHA-RL3c)@ @[Testable]@ If the room is in the @RELEASING@ status, then the operation will return the result of the pending @release@ operation, or throw any exception that it throws.
+** @(CHA-RL3c)@ @[Testable]@ When the release operation commences, the room will transition into the @RELEASING@ status any any transient disconnect timeouts shall be cleared.
+** @(CHA-RL3d)@ @[Testable]@ All features channels are detached in sequence.
+** @(CHA-RL3e)@ @[Testable]@ If a channel is in the @FAILED@ state when it is time to detach, it shall be ignored.
+** @(CHA-RL3f)@ @[Testable]@ If a channel fails to detach, the channel detach sequence will be retried after a short wait.
+** @(CHA-RL3g)@ @[Testable]@ Once all channels have entered the @DETACHED@ state, then the room state is transitioned to @RELEASED@ and the operation completes.
+** @(CHA-RL3h)@ @[Testable]@ Upon operation completion, the underlying Realtime Channels are released from the core SDK prevent leakage.
+* @(CHA-RL5)@ A room must @RETRY@ whenever it enters the @SUSPENDED@ state.
+** @(CHA-RL5a)@ @[Testable]@ When entering a retry-loop, the room must first @DETACH@ all contributors underlying realtime channels.
+** @(CHA-RL5b)@ @[Testable]@ If the operation above fails, then it must be retried after a short wait.
+** @(CHA-RL5c)@ @[Testable]@ If the operation above fails because a channel has entered the @FAILED@ state, then the retry loop must stop and the room must be placed in the @FAILED@ status.
+** @(CHA-RL5d)@ Once all channels have been detached, the room waits until the original channel that caused the retry loop naturally enters the @ATTACHED@ state.
+** @(CHA-RL5e)@ @[Testable]@ If the channel state becomes @FAILED@, then the room status is transitioned to @FAILED@ and the retry loop stops.
+** @(CHA-RL5f)@ @[Testable]@ If the channel state becomes @ATTACHED@, then the room status is transitioned to @ATTACHING@ to begin a new attachment cycle.
```

Comment:
The term "attachment cycle" is not defined and I am struggling to even guess what it means. I thought that perhaps this spec point was basically just saying "perform the ATTACH operation again", but that wouldn't make sense because we'd end up in deadlock (because CHA-RL1d would prevent ATTACH from proceeding due to the in-progress RETRY). So I guess it means something else?

Reply:
Sorry I will clarify on this one - essentially it's "begin the actual process of attaching, but without the need to lock a mutex to prevent other operations". In chat-js it's essentially just calling `_doAttach` for reference

Reply:
I think that if the idea is that it do a bunch of the same stuff that the ATTACH operation does, then we should probably either refer to exactly _which_ ATTACH behaviour spec points should be implemented (for example, it’s not clear to me whether we’re meant to do CHA-RL1h5’s detach-all-non-FAILED contributors).

Reply:
Yes, Ive added this

Reply:
So the CHA-RL1h5 detach is meant to happen as part of the RETRY too?

Reply:
I've removed CHA-RL5f1-3, as they're just re-stating CHA-RL1e+.